### PR TITLE
Lightspeed UI feedback and sentiment

### PR DESCRIPTION
**Pull request by ganeshrn**
*  Add  Ansible lightspeed feedback view (side panel) and webview (in editor tab) to get user feedback
*  Adds sentiment ratings and comment description box
*  For more feedback adds a drop down for `Bug report`, `Feature request` and `Suggestion Improvements`
* Update feedback API to throw an user error when user in not logged in.

feedback_1

feedback_2

feedback_3

feedback_4

### CODE REVIEW

**Review by priyamsahoo — APPROVED** [review]

**Review by ajinkyau — APPROVED** [review]
LGTM, but the UI tests are failing and checking the same with `vscode-extension-tester` team

File: src/webview/apps/lightspeed/main.ts [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+        <section class="component-section">
+          <vscode-button id="issue-submit-button">Submit</vscode-button>
+        </section>
+      `;
+      feedbackForm.appendChild(featureRequestSection);
+    } else if (selectedValue === "suggestion-feedback") {
+      suggestionFeedbackSection.innerHTML = `
+        <section class="component-section" class="issue-dropdown">
+          <p class="required">Prompt</p>
+          <vscode-text-area id="suggestion-prompt" cols="29" class="m-b-10" placeholder="Copy and Paste the file content till the end of task name description" resize="both" />
```
Comment (scottharwell) [inline src/webview/apps/lightspeed/main.ts:78]: Please change placeholder text to...

> Copy and paste the contents of the playbook until the name of the task used for a recommendation.

File: src/webview/apps/lightspeed/main.ts [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        </section>
+        <section class="component-section">
+          <p class="required">Provided Suggestion</p>
+          <vscode-text-area id="suggestion-provided" cols="29" class="m-b-10" placeholder="Provided Suggestion by Ansible Lightspeed" resize="both" />
+        </section>
+        <section class="component-section">
+          <p class="required">Expected Suggestion</p>
+          <vscode-text-area id="suggestion-expected" cols="29" class="m-b-10" placeholder="Your Expected Suggestion" resize="both" />
```
Comment (scottharwell) [inline src/webview/apps/lightspeed/main.ts:86]: The recommendation that you expected -- edit the response with what you expected the result to be

File: src/webview/apps/lightspeed/main.ts [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+        </section>
+        <section class="component-section">
+          <p class="required">Why was modification required?</p>
+          <vscode-text-area id="suggestion-additional-comment" cols="29" class="m-b-10" placeholder="Please enter details" resize="both" />
```
Comment (scottharwell) [inline src/webview/apps/lightspeed/main.ts:90]: We should remove "please" in all placeholders for brevity

Comment (ganeshrn) [reply src/webview/apps/lightspeed/main.ts:90]: fixed

File: src/webview/apps/lightspeed/main.ts (hunk 2 above)
Comment (ganeshrn) [reply src/webview/apps/lightspeed/main.ts:86]: fixed

File: src/webview/apps/lightspeed/main.ts [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+      feedbackForm.appendChild(featureRequestSection);
+    } else if (selectedValue === "suggestion-feedback") {
+      suggestionFeedbackSection.innerHTML = `
+        <section class="component-section" class="issue-dropdown">
+          <p class="required">Prompt</p>
+          <vscode-text-area id="suggestion-prompt" cols="29" class="m-b-10" placeholder="Copy and Paste the file content till the end of task name description" resize="both" />
+        </section>
+        <section class="component-section">
+          <p class="required">Provided Suggestion</p>
+          <vscode-text-area id="suggestion-provided" cols="29" class="m-b-10" placeholder="Provided Suggestion by Ansible Lightspeed" resize="both" />
```
[the remark this answers is not in the document]
Comment (ganeshrn) [reply src/webview/apps/lightspeed/main.ts:82]: fixed

File: src/webview/apps/lightspeed/main.ts (hunk 1 above)
Comment (ganeshrn) [reply src/webview/apps/lightspeed/main.ts:78]: fixed

**Pull request merged**